spack-repo/packages/singularity-eos/package.py

```diff
@@ -186,6 +193,14 @@ def cmake_args(self):
         #TODO: do we need this?
         if "+kokkos+cuda" in self.spec:
             args.append(self.define("CMAKE_CXX_COMPILER", self.spec["kokkos"].kokkos_cxx))
+        
+        if self.spec.satisfies("+plugins"):
+            PLUGIN_PATHS=os.environ.get("SINGULARITY_PLUGINS", "")
+            SINGULARITY_VARIANT=os.envirn.get("SINGULARITY_VARIANT", "")
+            if len(PLUGIN_PATHS) > 0:
+                args.append(self.define("SINGULARITY_PLUGINS", PLUGIN_PATHS))
+            if len(SINGULARITY_VARIANT) > 0:
+                args.append(self.define("SINGULARITY_VARIANT", SINGULARITY_VARIANT))
```

Comment:
Who sets the environment variables and when? Where does the plugins source code live? Instead of a Boolean variant, I would also imagine `plugins` to be something similar to `patches`.

```bash
spack install singularity-eos plugins=dir1:dir2:dir3 variant=some_path/to/another/file
```

I'm not sold on the idea to have this generalized in the public spackage. For the particular use case in mind, what's the downside of just having a inherited Spack package for singularity-eos and manipulating the CMake build and source directory for a particular plugin combo?

As in, Project XYZ has its own `singularity-eos` spackage that inherits from the upstream one, but grabs more source files, puts it in the right place and uses the CMake machinery to make it show up in the Singularity-EOS build.

Reply:
Do variants accept arbitrary strings? The reason I went with the environment variable path is that it wasn't clear from the spack documentation that this is possible. It is possible?

> I'm not sold on the idea to have this generalized in the public spackage. For the particular use case in mind, what's the downside of just having a inherited Spack package for singularity-eos and manipulating the CMake build and source directory for a particular plugin combo?

This is fine with me. The advantage of doing it here is that we only have to maintain one spackage. But maybe that's basically the case with the inheritence version as well? Since presumably the inherited package doesn't contain much?

Reply:
Ok I will go the spackage inheritence path for now.

Reply:
I'm not aware of a limitation of variants, they can be arbitrary strings. That does not mean they should be arbitrary complex strings. From my point of view, the main issue here is that spack covers runtime plugin systems (e.g. python/pypi), not compile time plugin systems.

Reply:
:+1: Either way is fine with me, so long as we have something that is useful broadly.